addons/im_livechat/models/chatbot_script_step.py

```diff
@@ -358,10 +358,14 @@ def _process_step_forward_operator(self, discuss_channel):
                 posted_message = discuss_channel._chatbot_post_message(self.chatbot_script_id, plaintext2html(self.message))
 
             # then post a small custom 'Operator has joined' notification
-            discuss_channel._chatbot_post_message(
-                self.chatbot_script_id,
-                Markup('<div class="o_mail_notification">%s</div>')
-                % _('%s has joined', human_operator.livechat_username or human_operator.partner_id.name))
+            discuss_channel.message_post(
+                author_id=human_operator.partner_id.id,
+                body=Markup('<div class="o_mail_notification">%s</div>') % _(
+                    '%s has joined',
+                    human_operator.livechat_username or human_operator.partner_id.name
+                ),
+                subtype_xmlid='mail.mt_comment'
+            )
```

Comment:
We can remove this message post and rely on the standard discuss flows instead by removing `post_joined_message=False` from line 348. Note that you need to use `with_user(human_operator)` to generate the desired message: "XXX joined the channel."

Fewer lines and reduced live chat-specific behavior, it's a win-win.

Reply:
@tsm-odoo  I think this might interrupt with the flow of how the operator joining notification is posted in the channel. With this, the operator's joining notification would now appear before any `self.message`, if present.

IMO we can first post the remaining `self.messages` and then add the operator to preserve the current flow.

Reply:
Agreed, switching the order looks good to me.